File: tests/nodejs/cases/case_224_ephemeral_lambda_use_02.js
/**
 * Task case 224: ephemeral_lambda_use_02 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case224EphemeralLambdaUse02(input) {
  const taskName = 'ephemeral_lambda_use_02';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const f = () => payload.task;
  void f();
  // SAFE: lambda used immediately and discarded.
  return payload.checksum;
}

module.exports = { case224EphemeralLambdaUse02 };

File: tests/go/cases/case_224_ephemeral_lambda_use_02.go
package escape_tests

var retainedCase224 = []map[string]string{}

func Case224EphemeralLambdaUse02(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "ephemeral_lambda_use_02",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "ephemeral_lambda_use_02:" + raw,
	}
	f := func() string { return payload["task"] }
	_ = f()
	// SAFE: immediate lambda usage only.
	return payload["checksum"]
}
